test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Import/ImportTests.cs

```diff
@@ -183,14 +183,119 @@ public async Task GivenIncrementalLoad_80KSurrogateIds_BadRequestIsReturned()
         }
 
         [Fact]
-        public async Task GivenIncrementalLoad_MultipleInputVersionsOutOfOrderSomeNotExplicit_ResourceNotExisting_NoGap()
+        public async Task GivenIncrementalLoad_MultipleInputsWithImplicitVersionsExplicitLastUpdatedBeforeImplicit()
+        {
+            var id = Guid.NewGuid().ToString("N");
+            var ndJson1 = CreateTestPatient(id);
+            var ndJson2 = CreateTestPatient(id, DateTimeOffset.UtcNow.AddDays(-1));
+            var location = (await ImportTestHelper.UploadFileAsync(ndJson1 + ndJson2, _fixture.StorageAccount)).location;
+            var request = CreateImportRequest(location, ImportMode.IncrementalLoad, false, true);
+            await ImportCheckAsync(request, null, 0);
+
+            var history = await _client.SearchAsync($"Patient/{id}/_history");
+            Assert.Single(history.Resource.Entry);
+            Assert.Equal("1", history.Resource.Entry[0].Resource.VersionId);
+        }
+
+        [Fact]
+        public async Task GivenIncrementalLoad_MultipleInputsWithImplicitVersionsExplicitLastUpdatedAfterImplicit()
+        {
+            var id = Guid.NewGuid().ToString("N");
+            var ndJson1 = CreateTestPatient(id);
+            var ndJson2 = CreateTestPatient(id, DateTimeOffset.UtcNow.AddDays(1));
+            var location = (await ImportTestHelper.UploadFileAsync(ndJson1 + ndJson2, _fixture.StorageAccount)).location;
+            var request = CreateImportRequest(location, ImportMode.IncrementalLoad, false, true);
+            await ImportCheckAsync(request, null, 0);
+
+            var history = await _client.SearchAsync($"Patient/{id}/_history");
+            Assert.Equal(2, history.Resource.Entry.Count);
+            Assert.Equal("1", history.Resource.Entry[0].Resource.VersionId);
+            Assert.Equal("-1", history.Resource.Entry[1].Resource.VersionId);
```

Comment:
Why doesn't this import as versions 1 & 2?

Reply:
Record with explicit last updated is loaded first, It takes version = 1. BTW It is from future. When record with implicit last updated is worked on, version 2 cannot be used because it would break our rule "version on current record is always max, so we can just increment in normal circumstances", Therefore -1.

Reply:
@SergeyGaluzo - should this be true even if the content of the resource is the same (except for explicit vs implicit last updated)? What would the behavior be if these resources were imported "in order" or "not late"?